src/array.jl

```diff
@@ -383,6 +383,17 @@ end
     @inbounds A.refs[I...] = get!(A.pool, v)
 end
 
+function Base.fill(v::CategoricalValue{T, R}, dims::NTuple{N, Integer}) where {T, R, N}
+    A = CategoricalArray{T, N, R}(undef, dims)
+    merge_pools!(A, v, updaterefs=false)
```

Comment:
can you please remind me what `updaterefs` does?

Reply:
It recomputed all reference codes to match the new levels order. Here since the contents are uninitialized (all zero), recomputing them would have no effect and just waste time.